crates/polars-ops/src/series/ops/round.rs

```diff
@@ -70,6 +107,14 @@ pub trait RoundSeries: SeriesSealed {
 
 impl RoundSeries for Series {}
 
+fn round_sig_figs(value: f64, digits: u32) -> f64 {
+    if value == 0.0 {
+        return value;
+    }
+    let magnitiude = 10.0.pow(digits as f64 - 1.0 - ((value.abs()).log10().floor()));
```

Comment:
Could you use `f64::powi`? Also it's spelled `magnitude`.

Reply:
i'm struggling to use the with_match_physical_numeric_polars_type macro. I cannot use $T to convert primitives so am not sure what the structure should look like to accomodate this.

```
        with_match_physical_numeric_polars_type!(s.dtype(), |$T| {
            let ca: &ChunkedArray<$T> = s.as_ref().as_ref().as_ref();
            if $T.is_float() {
                let s = ca.apply_values(|val| {
                    if val == 0.0 {
                        return 0.0;
                    }
                    let magnitude = 10.0_f64.powi(digits - 1 - ((value.abs()).log10().floor() as i32));
                    (value * magnitude).round() / magnitude
                }).into_series();
            } else {
                let s = ca.apply_values(|val| {
                    if val == 0 {
                        return 0;
                    }
                    let magnitude = 10.powi(digits - 1 - ((value.abs()).log10().floor() as i32));
                    (value * magnitude).round() / magnitude
                }).into_series();
            }

            return Ok(s);
        });
```

Reply:
Think this is done.